lib/datadog/appsec/response.rb

```diff
@@ -0,0 +1,50 @@
+# typed: false
+
+require_relative 'assets'
+
+module Datadog
+  module AppSec
+    # AppSec response
+    class Response
+      attr_reader :status, :headers, :body
+
+      def initialize(status:, headers: {}, body: [])
+        @status = status
+        @headers = headers
+        @body = body
+      end
+
+      def to_rack
+        [status, headers, body]
+      end
+
+      def to_sinatra_response
+        ::Sinatra::Response.new(body, status, headers)
+      end
+
+      def to_action_dispatch_response
+        ::ActionDispatch::Response.new(status, headers, body)
+      end
+
+      def self.negotiate(env)
+        format = if accepted?('text/html', env)
+                   :html
+                 elsif accepted?('application/json', env)
+                   :json
+                 else
+                   :text
+                 end
+
+        Response.new(status: 403,
+                     headers: { 'Content-Type' => 'text/html' },
+                     body: [Datadog::AppSec::Assets.blocked(format: format)])
+      end
+
+      private
+
+      def self.accepted?(mime, env)
+        env['HTTP_ACCEPT'] && env['HTTP_ACCEPT'].split(',').any? { |e| e.start_with?(mime) }
+      end
```

Comment:
Would you mind refactoring this to avoid splitting the header string multiple times? A suggestion:
```suggestion
      def self.negotiate(env)
        Response.new(status: 403,
                     headers: { 'Content-Type' => 'text/html' },
                     body: [Datadog::AppSec::Assets.blocked(format: format(env))])
      end

      private

      def self.format(env)
        format = env['HTTP_ACCEPT'] && env['HTTP_ACCEPT'].split(',').any? do |accept|
          if accept.start_with?('text/html')
            :html
          elsif accept.start_with?('application/json')
            :json
          end
        end

        format || :text
      end
```

Reply:
Since this is not a fully compliant content-negotiation implementation, I was trying to keep response content type negotiation separate from negotiated response content type to format mapping.

Nonetheless, it makes sense to not process this crude content negotiation multiple times indeed.

Committing your suggestion.